java/adapter/avro/src/main/java/org/apache/arrow/consumers/AvroStringConsumer.java

```diff
@@ -28,9 +28,9 @@
  * Consumer which consume string type values from avro decoder.
  * Write the data to {@link VarCharVector}.
  */
-public class AvroStringConsumer implements Consumer {
+public class AvroStringConsumer implements Consumer<VarCharVector> {
```

Comment:
does making these Generic actually help in anyway?

Reply:
Now mainly used in resetValueVector, this could ensure the vector type is correct I think.

Reply:
I guess we can leave it for now.   but it essentially is the same thing as making resetValueVector take ValueVector and casting within the function.